tools/clang/lib/Sema/SemaChecking.cpp

```diff
@@ -48,6 +48,24 @@ SourceLocation Sema::getLocationOfStringLiteralByte(const StringLiteral *SL,
                                Context.getTargetInfo());
 }
 
+// HLSL Change begin
+static void CheckGloballyCoherentMismatch(Sema &S, Expr *SrcExpr,
+                                          QualType DstTy, SourceLocation CC) {
+  QualType SrcTy = SrcExpr->getType();
+  if (SrcTy->isArrayType() && DstTy->isArrayType()) {
+    SrcTy = SrcTy->getAsArrayTypeUnsafe()->getElementType();
```

Comment:
Do we want to support multi-dim resource array ?

Reply:
Fixed in b104eb4b7b15dc4c43eda3ecc8e128296cd4aef7